src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -809,9 +809,12 @@ namespace smt::noodler {
         bool contains_word_disequations = !this->m_word_diseq_todo_rel.empty();
         bool contains_conversions = !this->m_conversion_todo.empty();
 
-        // As a heuristic, for the case we have exactly one constraint, which is of type 'x (not)in RE', we use universality/emptiness
-        // checking of the regex (using some heuristics) instead of constructing the automaton of RE. The construction (especially complement)
-        // can sometimes blow up, so the check should be faster.
+        // Change name
+        bool contains_equations_only = this->m_lang_eq_or_diseq_todo_rel.empty() && this->m_not_contains_todo_rel.empty() && this->m_conversion_todo.empty();
```

Comment:
You do not need to check for `this->m_lang_eq_or_diseq_todo_rel.empty()`, language (dis)equations are handled before.

Reply:
Right; I fixed it.